devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy

```diff
@@ -61,12 +68,31 @@ def updated() {
 	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```

Comment:
The one-hour checkin is specific to the First Alert models. You'll need to determine if this device has a regular check-in and if not, use the 8 hour checkInterval that is default for sleepy z-wave devices.

In that case, make 8 hours the default, and set to the shorter 2 hours if the device is the FirstAlert model (use zwaveInfo.mfr to determine manufacturer)